src/libcollections/btree/map.rs

```diff
@@ -342,6 +342,14 @@ pub struct OccupiedEntry<'a, K: 'a, V: 'a> {
     _marker: PhantomData<&'a mut (K, V)>,
 }
 
+// An iterator for merging two sorted sequences into one
+struct MergeIter<K, V, I: Iterator<Item=(K, V)>> {
+    left: I,
+    right: I,
+    left_cur: Option<(K, V)>,
```

Comment:
Instead of storing `left_cur` directly, you could just store two `Peekable` iterators.